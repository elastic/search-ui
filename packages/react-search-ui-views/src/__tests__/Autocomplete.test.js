import React from "react";
import { Autocomplete } from "..";
import { shallow } from "enzyme";

const props = {
  autocompleteResults: {
    sectionTitle: "Results",
    titleField: "title",
    urlField: "nps_link",
    linkTarget: "_blank"
  },
  autocompletedResults: [
    {
      id: { raw: "1" },
      title: { snippet: "<em>Bike</em> Cops" }
    },
    {
      id: { raw: "2" },
      title: { snippet: "<em>Biker</em> Gang" }
    },
    {
      id: { raw: "3" },
      title: { snippet: "<em>Biker</em> Bar" }
    }
  ],
  autocompleteSuggestions: {
    documents: {
      sectionTitle: "Suggested"
    },
    popular_queries: {
      sectionTitle: "Popular"
    }
  },
  autocompletedSuggestions: {
    documents: [
      { highlight: "", suggestion: "bike" },
      { highlight: "", suggestion: "bike police" },
      { highlight: "", suggestion: "bike police go" }
    ],
    popular_queries: [
      {
        highlight: "",
        suggestion: "how do i know when my bike needs new tires?"
      },
      { highlight: "", suggestion: "what is a banana bike?" },
      { highlight: "", suggestion: "is it cool to ride a bike?" }
    ]
  },
  getItemProps: props => props,
  getMenuProps: props => props
};

it("renders correctly", () => {
  const wrapper = shallow(<Autocomplete {...props} />);
  expect(wrapper).toMatchSnapshot();
});
