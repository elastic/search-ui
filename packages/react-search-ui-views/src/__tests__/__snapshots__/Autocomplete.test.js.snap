// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sui-search-box__autocomplete-container"
>
  <div>
    <div
      className="sui-search-box__section-title"
    >
      Suggested
    </div>
    <ul
      className="sui-search-box__suggestion-list"
    >
      <li
        index={0}
        item={
          Object {
            "highlight": "",
            "suggestion": "bike",
          }
        }
        key="bike"
      >
        <span>
          bike
        </span>
      </li>
      <li
        index={1}
        item={
          Object {
            "highlight": "",
            "suggestion": "bike police",
          }
        }
        key="bike police"
      >
        <span>
          bike police
        </span>
      </li>
      <li
        index={2}
        item={
          Object {
            "highlight": "",
            "suggestion": "bike police go",
          }
        }
        key="bike police go"
      >
        <span>
          bike police go
        </span>
      </li>
    </ul>
    <div
      className="sui-search-box__section-title"
    >
      Popular
    </div>
    <ul
      className="sui-search-box__suggestion-list"
    >
      <li
        index={3}
        item={
          Object {
            "highlight": "",
            "suggestion": "how do i know when my bike needs new tires?",
          }
        }
        key="how do i know when my bike needs new tires?"
      >
        <span>
          how do i know when my bike needs new tires?
        </span>
      </li>
      <li
        index={4}
        item={
          Object {
            "highlight": "",
            "suggestion": "what is a banana bike?",
          }
        }
        key="what is a banana bike?"
      >
        <span>
          what is a banana bike?
        </span>
      </li>
      <li
        index={5}
        item={
          Object {
            "highlight": "",
            "suggestion": "is it cool to ride a bike?",
          }
        }
        key="is it cool to ride a bike?"
      >
        <span>
          is it cool to ride a bike?
        </span>
      </li>
    </ul>
    <div
      className="sui-search-box__section-title"
    >
      Results
    </div>
    <ul
      className="sui-search-box__results-list"
    >
      <li
        index={6}
        item={
          Object {
            "id": Object {
              "raw": "1",
            },
            "title": Object {
              "snippet": "<em>Bike</em> Cops",
            },
          }
        }
        key="1"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<em>Bike</em> Cops",
            }
          }
        />
      </li>
      <li
        index={7}
        item={
          Object {
            "id": Object {
              "raw": "2",
            },
            "title": Object {
              "snippet": "<em>Biker</em> Gang",
            },
          }
        }
        key="2"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<em>Biker</em> Gang",
            }
          }
        />
      </li>
      <li
        index={8}
        item={
          Object {
            "id": Object {
              "raw": "3",
            },
            "title": Object {
              "snippet": "<em>Biker</em> Bar",
            },
          }
        }
        key="3"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<em>Biker</em> Bar",
            }
          }
        />
      </li>
    </ul>
  </div>
</div>
`;
